website/pages/blog/optimize-bundle-size-with-swc-and-graphql-codegen.mdx

```diff
@@ -0,0 +1,191 @@
+---
+title: Optimize your Bundle Size with SWC and GraphQL Codegen
+authors: jessevelden
+tags: [client-preset, codegen]
+date: 2023-02-24
+description:
+  'Optimize your Bundle Size with the SWC plugin for client-preset for GraphQL Code Generator'
+image: /blog-assets/optimize-bundle-size-with-swc-and-graphql-codegen/cover.png
+thumbnail: /blog-assets/optimize-bundle-size-with-swc-and-graphql-codegen/thumbnail.png
+---
+
+import { Callout } from '@theguild/components'
+
+With the new
+[`client-preset` for GraphQL Code Generator](https://the-guild.dev/blog/unleash-the-power-of-fragments-with-graphql-codegen)
+it is possible to write your GraphQL queries inlined, next to your component in a type-safe way.
+This can greatly improve the Developer Experience (DX).
+
+Instead of writing your operations in a separate file, you can write them inline in your component,
+and use the `graphql` function to parse the operation and get the correct types.
+
+```tsx
+// Instead of operations in a separate .graphql file and importing generated hooks:
+import { useGetCartByIdQuery } from './graphql/generated-hooks'
+const { data } = useGetCartByIdQuery({ variables: { id: '1' } })
+
+// You can write your operations directly next to your component and get the correct types:
+import { graphql } from './graphql'
+const GetCartByIdDocument = graphql(/** GraphQL */ `
+  query GetCartById($id: ID!) {
+    id
+    items {
+      id
+      ...CartItem
+    }
+  }
+`)
+// Awesome! We have the correct types for this operation even when we specify it next to our component!
+```
+
+However, this approach can come at a cost - the bundle size of your pages can increase
+significantly, especially as the number of queries and mutation grows. In this post, we will show
+you what the problem is and how to optimize your bundle size with the new
+[`@graphql-codegen/client-preset-swc-plugin`](https://www.npmjs.com/package/@graphql-codegen/client-preset-swc-plugin)
+plugin for your applications supported by the SWC compiler such as projects using
+[Next.js](https://nextjs.org) or [Vite](https://vitejs.dev/).
+
+## The Generated Code When Using the `client-preset`
+
+When using the `client-preset` without any additional optimization, your site's bundle can quickly
```

Comment:
@saihaj @n1ru4l @YassinEldeeb We have an open issue about that to solve it for others as well right?

Should we link to it here? otherwise it's very negative for people not using Next/Vite?

Reply:
oh I see, for people using Babel there is a solution.
I thin it's important to highlight it here so people won't think they don't have a solution.
also, what is the use case for people not using Babel and SWC?

Reply:
> oh I see, for people using Babel there is a solution. I thin it's important to highlight it here so people won't think they don't have a solution. also, what is the use case for people not using Babel and SWC?

I guess people who don't want an extra build step, but that seems to be rare now a days. If you're using GraphQL Codegen you're already introducing an extra build step, so adding one more to optimize it should be not that inconvenient.